packages/dds/tree/src/simple-tree/objectNode.ts

```diff
@@ -155,79 +133,71 @@ function createObjectProxy(
 
 	// TODO: Although the target is an object literal, it's still worthwhile to try experimenting with
 	// a dispatch object to see if it improves performance.
-	const proxy = new Proxy(targetObject, {
-		get(target, viewKey): unknown {
-			const flexKey = getFlexKey(viewKey as FieldKey);
-			const field = getFlexNode(proxy).tryGetField(flexKey);
+	const handler: ProxyHandler<TreeNode> = {
+		get(target, viewKey, proxy): unknown {
+			const fieldInfo = flexKeyMap.get(viewKey);
 
-			if (field !== undefined) {
-				return getProxyForField(field);
+			if (fieldInfo !== undefined) {
+				const field = getFlexNode(proxy).tryGetField(fieldInfo.storedKey);
+				return field === undefined ? undefined : getProxyForField(field);
 			}
 
 			// Pass the proxy as the receiver here, so that any methods on the prototype receive `proxy` as `this`.
 			return Reflect.get(target, viewKey, proxy);
 		},
-		set(target, viewKey, value: InsertableContent) {
-			const flexNode = getFlexNode(proxy);
-			const flexNodeSchema = flexNode.schema;
-			assert(flexNodeSchema instanceof FlexObjectNodeSchema, 0x888 /* invalid schema */);
-
-			const hasField = viewKey in schema.info;
-
-			if (!hasField) {
+		set(target, viewKey, value: InsertableContent, proxy) {
+			const fieldInfo = flexKeyMap.get(viewKey);
+			if (fieldInfo === undefined) {
 				return allowAdditionalProperties ? Reflect.set(target, viewKey, value) : false;
 			}
-			assert(typeof viewKey === "string", 0x7e1 /* invalid key */);
-			const flexKey: FieldKey | undefined = getFlexKey(viewKey);
-
-			const field = flexNode.getBoxed(flexKey);
-
-			const simpleNodeFields = schema.info;
-			if (simpleNodeFields[viewKey] === undefined) {
-				fail(`Field key '${viewKey}' not found in schema.`);
-			}
 
-			const simpleFieldSchema = getSimpleFieldSchema(field.schema, schema.info[viewKey]);
-
-			setField(field, simpleFieldSchema, value);
+			const flexNode = getFlexNode(proxy);
+			const field = flexNode.getBoxed(fieldInfo.storedKey);
+			setField(field, fieldInfo.schema, value);
 
 			return true;
 		},
 		has: (target, viewKey) => {
-			const fields = schema.info as Record<string, ImplicitFieldSchema>;
 			return (
-				fields[viewKey as FieldKey] !== undefined ||
+				flexKeyMap.has(viewKey) ||
 				(allowAdditionalProperties ? Reflect.has(target, viewKey) : false)
 			);
 		},
 		ownKeys: (target) => {
-			const fields = schema.info as Record<string, ImplicitFieldSchema>;
 			return [
-				...Object.keys(fields),
+				...flexKeyMap.keys(),
```

Comment:
Reading https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/keys it seems this is only supposed to include enumerable keys.

Thus this code was wrong before (included non existent keys), and is now less wrong (includes non-enumerable keys.

At least has should be correct now.

Reply:
Actually https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy/Proxy/ownKeys doesn't say this should only be enumerable keys, so maybe this is file. Object.keys might filter. I'll add a test.

Reply:
Test added. Object.keys works as documented excluding non enumerable fields even though they are included here, so everything should be good.